[next] Fixes issue#1481: re-implements the LogoIcon component

chrispinkney opened the pull request:
Fixes #1481:
* Added LogoIcon component and the `.svg` Telescope logo.

<!--
Thanks for sending a pull request!
- if this is your first time, please ensure you have read through our contributor guide: https://github.com/Seneca-CDOT/telescope/blob/master/docs/CONTRIBUTING.md
-->

## Issue This PR Addresses

<!--
1. Automatically close the issue when this PR is merged
    USAGE: Fixes #<issue number>
2. If your PR addresses an issue but does not close it
    USAGE: #<issue number> <reason>(i.e. This issue was worked on by @user and myself and his PR should close the issue.)
-->

## Type of Change

<!-- bug fix, feature, documentation, UI, etc. -->

- [ ] **Bugfix**: Change which fixes an issue
- [X] **New Feature**: Change which adds functionality
- [ ] **Documentation Update**: Change which improves documentation
- [ ] **UI**: Change which improves UI

## Description

Re-implements the `LogoIcon` component in next.js using a webp version of the logo (because, why not? _but seriously I can change it back if required, I was just curious about the webp format_).

Simply import and call LogoIcon while specifying a size and we're off to the races:

```TS
import LogoIcon from './LogoIcon';
<LogoIcon height={320} width={320} />
```

![2020-12-09_22-24-48](https://user-images.githubusercontent.com/7242003/101721094-29b14480-3a75-11eb-80ff-08380ec54d9c.gif)

<!-- Please add a detailed description of what this PR does and why it is needed -->

## Checklist

<!-- Before submitting a PR, address each item -->

- [ ] **Quality**: This PR builds and passes our npm test and works locally
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [X] **Screenshots**: This PR includes screenshots or GIFs of the changes made or an explanation of why it does not (if applicable)
- [ ] **Documentation**: This PR includes updated/added documentation to user exposed functionality or configuration variables are added/changed or an explanation of why it does not(if applicable)

tonyvugithub commented on src/frontend/next/src/pages/LogoIcon.tsx:
```diff
@@ -0,0 +1,14 @@
+    </>
+  );
+};
+
```
Maybe create an interface for props type and call it LogoIconProps
```
import {FC} from 'react';
const interface LogoIconProps {
    height: number,
    width: number,
}
const LogoIcon: FC<LogoIconProps> = ({height, width}) => {
    .....
}
```

tonyvugithub replied:
FC stands for functional component that would take in props in the form of LogoIconProps

chrispinkney replied:
Kk on it!

humphd reviewed (changes requested):
Webp is not supported in all browsers.  If you want to do this, you need to use `<picture>` and provide different sources, with a fallback to JPG.

I'm not sure that webp gains us much here with such a small image.

Also, if we put the files in `public/` don't we get to use them for free without needing `next-images` and config changes?

chrispinkney commented:
I'm getting some eslint errors when trying to import my component and I'm not sure why- although the component does import and works as expected. :shrug:

humphd commented on src/frontend/next/src/pages/LogoIcon.tsx:
```diff
@@ -0,0 +1,18 @@
+import { FC } from 'react';
+import Image from 'next/image';
```
```
7:25:05 PM: Error: Image Optimization using Next.js' default loader is not compatible with `next export`.
Possible solutions:
7:25:05 PM:   - Use `next start`, which starts the Image Optimization API.
7:25:05 PM:   - Use Vercel to deploy, which supports Image Optimization.
7:25:05 PM:   - Configure a third-party loader in `next.config.js`.
7:25:05 PM: Read more: https://err.sh/next.js/export-image-api
7:25:05 PM:     at exportApp (/opt/build/repo/src/frontend/next/node_modules/next/dist/export/index.js:14:712)
```

We can't use `Image` unfortunately, it only works with server-side optimization (ie., you have to run a server to create dynamic images on demand).  For pure static/exported projects, it's not possible.

chrispinkney replied:
That's too bad. Ah well, I removed it.

humphd commented on src/frontend/next/src/pages/LogoIcon.tsx:
```diff
@@ -0,0 +1,18 @@
+import Image from 'next/image';
+
+// This interface specifies that the props passed in can only be of type number
+interface LogoIconProps {
```
This should be `type` vs. `interface`

chrispinkney replied:
Done!

humphd reviewed (changes requested):
Why have you removed next.config.js?

chrispinkney commented:
> 
> 
> Why have you removed next.config.js?

@humphd I was using the `next-images` library and because of that I had to create the file, but since I just removed the library I deleted the file as well as it added nothing but configuration for the library.

tonyvugithub commented on src/frontend/next/src/pages/LogoIcon.tsx:
```diff
@@ -0,0 +1,17 @@
+import { FC } from 'react';
+
```
@chrispinkney : The path should be `/src/components/LogoIcon/LogoIcon.tsx`. Also can you squash into 1 commit instead of 2?

chrispinkney replied:
Done like dinner.

humphd reviewed (approved):
The FC and file layout can be fixed later if you want, this is good otherwise.
